File: Java/BinarySearch.java
public class BinarySearch {
    public static void main(String[] args) {
        System.out.println("Binary Search....!!!!!!");
        int[] arr = {1,2,3,4,5,7,8,9,13,14,16,19,21,24,26,28,29,35,37,45,68,78,89,90,100};
        System.out.println(BS(arr,16));
    }
    static int BS(int[] arr,int target){
        int start = 0;
        int end = arr.length -1;
        while(start <= end){
            int mid = start + (end -start) / 2;
            if(arr[mid] < target){
                start = mid + 1;
            }
            else if(arr[mid] > target){
                end = mid - 1;
            }
            else {
                return mid;
            }
        }
        return start;
    }
}

File: TypeScript/binarySearch.ts
interface Params{
    arr:number[],
    target:number,
}

const arr:number[] = [2,4,6,8];
const target:number = 6;



function binarySearch({arr,target}:Params){
    var start:number = 0;
    var mid:number = 0;
    var end:number = arr.length -1;

    while(start<end){
        mid = (start+end)/2;
        if(arr[mid] === target){
            return mid;
        }

        if(arr[mid] > target){
            end = mid;
        }else{
            start = mid +1;
        }
    }
    return start;
}

console.log(binarySearch({arr,target}));